frontend/components/Filter/StudentFilter/StudentFilter.tsx

```diff
@@ -0,0 +1,561 @@
+    const [studentCoach, setstudentCoach] = useState<boolean>(false);
+    const [statusFilter, setStatusFilter] = useState<string>("");
+    const [osocYear, setOsocYear] = useState<string>("");
+    const [emailNone, setEmailNone] = useState<boolean>(false);
+    const [emailDraft, setEmailDraft] = useState<boolean>(false);
+    const [emailSent, setEmailSent] = useState<boolean>(false);
+    const [emailFailed, setEmailFailed] = useState<boolean>(false);
+    const [emailScheduled, setEmailScheduled] = useState<boolean>(false);
+    const [statusEmail, setStatusEmail] = useState<string>("");
+
+    // Roles used in the dropdown
+    const [roles, setRoles] = useState<Array<Role>>([]);
+    // A set of active roles
+    const [selectedRoles, setSelectedRoles] = useState<Set<string>>(new Set());
+    const [dropdownActive, setDropdownActive] = useState<boolean>(false);
+
+    const fetchRoles = async () => {
+        const sessionKey =
+            getSessionKey != undefined ? await getSessionKey() : "";
+        const responseRoles = await fetch(
+            `${process.env.NEXT_PUBLIC_API_URL}/role/all`,
+            {
+                method: "GET",
+                headers: {
+                    Authorization: `auth/osoc2 ${sessionKey}`,
+                },
+            }
+        )
+            .then((response) => response.json())
+            .then((json) => {
+                if (!json.success) {
+                    return { success: false };
+                } else return json;
+            })
+            .catch((err) => {
+                console.log(err);
+                return { success: false };
+            });
+        if (setSessionKey) {
+            setSessionKey(responseRoles.sessionkey);
+        }
+        setRoles(responseRoles.data);
+    };
+
+    useEffect(() => {
+        if (roles.length === 0) {
+            fetchRoles().then(() => search());
+        } else {
+            search().then();
+        }
+    }, [
+        firstNameSort,
+        lastNameSort,
+        emailSort,
+        alumni,
+        studentCoach,
+        statusFilter,
+    ]);
+
+    const toggleFirstNameSort = async (e: SyntheticEvent) => {
+        e.preventDefault();
+        setFirstNameSort((prev) => getNextSort(prev));
+    };
+
+    const toggleLastNameSort = async (e: SyntheticEvent) => {
+        e.preventDefault();
+        setLastNameSort((prev) => getNextSort(prev));
+    };
+
+    const toggleEmailSort = async (e: SyntheticEvent) => {
+        e.preventDefault();
+        setEmailSort((prev) => getNextSort(prev));
+    };
+
+    const toggleFilterYes = async (e: SyntheticEvent) => {
+        e.preventDefault();
+        //This is because the call is async
+        if (!filterYes) {
+            setStatusFilter(() => "YES");
+        } else {
+            setStatusFilter(() => "");
+        }
+        setFilterYes((bool) => !bool);
+        setFilterMaybe(false);
+        setFilterNo(false);
```

Comment:
Why disallow multiple filters? If you are only going to use 1 filter at a time you should just replace the 3 variables with an enum.